webapp/lib/app/configurator/standard_messages/controller.dart

```diff
@@ -90,76 +95,117 @@ class MessagesConfiguratorController extends ConfiguratorController {
     switch (action) {
       case MessagesConfigAction.addStandardMessage:
         StandardMessageData messageData = data;
-        var message = standardMessagesManager.createMessage(messageData.category, messageData.group);
-
-        _addMessagesToView({
-          message.category: {
-            message.groupDescription: [message]
-          }
-        });
+        var categoryName = standardMessagesManager.categories[messageData.categoryId].categoryName;
+        var groupName = standardMessagesManager.categories[messageData.categoryId].groups[messageData.groupId].groupName;
+
+        var message = standardMessagesManager.createMessage(messageData.categoryId, categoryName, messageData.groupId, groupName);
+        var messageCategoryMap = {
+          message.categoryId: MessageCategory(messageData.categoryId, categoryName)
+            ..groups = {
+              message.groupId: MessageGroup(messageData.groupId, groupName)
+                ..messages = {
+                  message.docId: message
+                }
+            }
+        };
+        _addMessagesToView(messageCategoryMap);
         _updateUnsavedIndicators(standardMessagesManager.categories, standardMessagesManager.unsavedMessageIds, standardMessagesManager.unsavedGroupIds, standardMessagesManager.unsavedCategoryIds);
         break;
 
       case MessagesConfigAction.updateStandardMessage:
         StandardMessageData messageData = data;
-        var standardMessage = standardMessagesManager.modifyMessage(messageData.id, messageData.text, messageData.translation);
-        _modifyMessagesInView({
-          standardMessage.category: {
-            standardMessage.groupDescription: [standardMessage]
+        var standardMessage = standardMessagesManager.modifyMessage(messageData.messageId, messageData.text, messageData.translation);
+        String categoryName;
+        String groupName;
+        for(var category in standardMessagesManager.categories.values) {
+          for(var group in category.groups.values) {
+            var messageIds = group.messages.keys;
+            if (messageIds.contains(messageData.messageId)) {
+              categoryName = category.categoryName;
+              groupName = group.groupName;
+              break;
+            }
           }
-        });
+        }
```

Comment:
I think this can be shortened to:
```
var categoryName = standardMessagesManager.categories[standardMessage.categoryId].categoryName;
var groupName = standardMessagesManager.categories[standardMessage.categoryId].groups[standardMessage.groupId].groupName;
```
(I think this works since we have the `standardMessage` which is a `model.SuggestedReply`?)